sw/device/lib/runtime/otbn.c

```diff
@@ -195,3 +196,16 @@ otbn_result_t otbn_zero_data_memory(otbn_t *ctx) {
   }
   return retval;
 }
+
+void otbn_dump_dmem(const otbn_t *ctx, uint32_t max_addr) {
+  if (max_addr == 0) {
+    max_addr = dif_otbn_get_dmem_size_bytes(&ctx->dif);
+  }
+
+  uint32_t data[8];
+  for (int i = 0; i < max_addr / 32; ++i) {
+    dif_otbn_dmem_read(&ctx->dif, i * 32, data, 32);
+    LOG_INFO("DMEM @%04d: 0x%08x%08x%08x%08x%08x%08x%08x%08x", i, data[7],
+             data[6], data[5], data[4], data[3], data[2], data[1], data[0]);
```

Comment:
Is there a reason for printing this in big-endian order?

Reply:
Each LOG_INFO is an individual 256b wide data word/number that are used as such in the crypto algorithms. We want a representation that makes it easy to compare the contents against, e.g. the simulator.